StRoot/StFcsDbMaker/StFcsDb.cxx

```diff
@@ -354,8 +373,31 @@ int StFcsDb::InitRun(int runNumber) {
       readGainCorrFromText();
     }
 
+    int ie=0, ih=0, ip=0, ehp, ns, crt, slt, dep, ch;
+    for(int ins=0; ins<kFcsNorthSouth; ins++){
+        int det=kFcsEcalNorthDetId+ins;
+        for(int id=0; id<maxId(det); id++){
+	    getDepfromId(det,id,ehp,ns,crt,slt,dep,ch);
+	    mGainOnline[ehp][ns][dep][ch]=mFcsEcalGainOnline.gainOnline[ie];
+	    ie++;
+        }
+        det=kFcsHcalNorthDetId+ins;
+        for(int id=0; id<maxId(det); id++){
+	    getDepfromId(det,id,ehp,ns,crt,slt,dep,ch);
+	    mGainOnline[ehp][ns][dep][ch]=mFcsHcalGainOnline.gainOnline[ih];
+	    ih++;
+        }
+        det=kFcsPresNorthDetId+ins;
+        for(int id=0; id<maxId(det); id++){
+	    getDepfromId(det,id,ehp,ns,crt,slt,dep,ch);
+	    mGainOnline[ehp][ns][dep][ch]=mFcsPresThreshold.threshold[ip];
+	    ip++;
+        }
+    }    
+
     // Get beamline 
-    TDataSet* dbDataSet = StMaker::GetChain()->GetDataBase("Calibrations/rhic/vertexSeed");
+    //TDataSet* dbDataSet = StMaker::GetChain()->GetDataBase("Calibrations/rhic/vertexSeed");
+    TDataSet* dbDataSet = 0;
```

Comment:
Hi Akio, could you please explain the redefinition of the dbDataSet variable here. With this definition the block of code below would not execute. I also don't see this in the code that I copied over from your local directory so I was curious about this change.

Reply:
This is not currenty used but here as placeholder for when significant crossing angle exist (like pAu, planned but not happening in run24). This cuased some crashes and this is walk around. Yes this looks bad but we need to resolve this when we needed.

Reply:
Great that addresses my question thank you for the update.